.github/workflows/ci-workflow.yml

```diff
@@ -1,32 +1,45 @@
+---
+
 name: Compilation & tests
 
 on:
   push:
-    branches: [ "master", "dev" ]
+    branches: ["master", "dev", "main", "develop"]
   pull_request:
-    branches: [ "master", "dev" ]
+    branches: ["master", "dev", "main", "develop"]
 
 jobs:
   job_build_debug:
     name: Build debug
+    strategy:
+      matrix:
+        device: [nanos, nanox, nanosp]
     runs-on: ubuntu-latest
 
     container:
       image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
 
     steps:
       - name: Clone
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       - name: Build
         run: |
-          make DEBUG=1
+          echo "Building for ${{ matrix.device }}..." && \
+          eval "BOLOS_SDK=\$$(echo ${{ matrix.device }} | tr [:lower:] [:upper:])_SDK" && \
+          echo "BOLOS_SDK value will be ${BOLOS_SDK}" && \
+          make BOLOS_SDK=${BOLOS_SDK} DEBUG=1
```

Comment:
`BOLOS_SDK` will be set to `NANOS_SDK`, `NANOX_SDK` or `NANOSP_SDK` depending on the matrix device.

Which will use one of the following in the builder container

```bash
bash-5.1# env
NANOSP_SDK=/opt/nanosplus-secure-sdk
NANOS_SDK=/opt/nanos-secure-sdk
BOLOS_SDK=/opt/nanos-secure-sdk
...
```